clients/python-wrapper/lakefs/client.py

```diff
@@ -106,6 +110,70 @@ def version(self) -> str:
         return self._server_conf.version
 
 
+def get_identity_token(session: 'boto3.Session') -> str:
+    """
+   Generate the identity token required for lakeFS authentication from an AWS session.
+
+   This function uses the STS client to generate a presigned URL for the `get_caller_identity` action, extracts the required values from the URL,
+   and creates a base64-encoded JSON object with these values.
+
+   :param session: A boto3 session object with the necessary AWS credentials and region information.
+   :return: A base64-encoded JSON string containing the required authentication information.
+   :raises ValueError: If the session does not have a region name set.
+   """
+
+    if session.region_name is None:
+        raise ValueError("Region name is not set in the session")
```

Comment:
There's an issue with separate region parameter like that.

**Context 1**: 
AWS has regional endpoints for sts, if the endpoint has no region than it's us-east-1 - [see](https://docs.aws.amazon.com/general/latest/gr/sts.html). 

**Context 2:** 
To use a regional endpoint (common in orgs with hardened security) you can use regional endpoint only when you set `AWS_STS_REGIONAL_ENDPOINTS=regional` then the sts_client will use the provided region for the host name. i.e `sts.eu-central-1.amazonaws.com` instead of `sts.amazonaws.com` depending on the region you provide when the session is initiated, **otherwise regardless of the region the sts endpoint will be `sts.amazonaws.com` which is default region `us-east-1`.** 

**Context 3**: 

In lakeFS when we recreate the pre-signed request the region is part of the scope. 
If the region in the scope (part of `X-Amz-Credential` header) != sts host region then you'll get an error from AWS `Credential should be scoped to a valid region`

```
https://sts.eu-central-1.amazonaws.com/\?Action\=GetCallerIdentity...X-Amz-Credential\=..eu-central-1...
```

Now what I suggest is similar to prior art (Vault and lakeFS HadoopFS client), infer the region from the sts endpoint. 

Instead of setting `"region": session.region_name`, infer from the `parsed_url.hostname` and no need in doing assert `if session.region_name is None:`

if it's `sts.amazonaws.com` -> `us-east-1`
else, sts.<REGION>.amazonaws.com -> `REGION`.

Reply:
If I understand you correctly, since the boto3 session encapsulates everything we should just let it handle the region detection and not error on it

Reply:
@N-o-Z The request that is built is sent to lakeFS, Boto handles this correctly, but lakeFS requires these fields in order to validate, @Isan-Rivkin correct me if I'm wrong.
The only thing I don't understand is wht does lakeFS require a region if it validates the host as well, isn't that just checking that the client knows how to extract the region? Anyway thanks for catching this, I'm fixing it

Reply:
So maybe we should fallback to `us-east-1` as default?

Reply:
Ended up letting Boto generate the URL and extracting the region from it, if it's `sts.amazonaws.com use` `us-east-1`